### ISSUE TITLE
All .pcap files have 0 bytes on flipper's SD card

### ISSUE THREAD

**Issue by b1scoito** [ElectronicCats/flipper-shields#3]
Hello, I've been trying to make .pcap files writing work with your guys board, but to no success, the onboard ESP32-S3-WROOM-1U doesn't come with a SD card slot and serial communication between ESP32Marauder and the Flipper Zero doesn't appear to work, Wi-Fi Marauder GPIO app only writes 0 bytes .pcap files, does the board support writing through serial, or is there any custom firmware for the Electronic Cats board, I've actually not tested the firmware that came with the board since I wanted to update to the latest ESP32Marauder as soon as possible. Have you guys tested the .pcap writing functionality under Apps -> GPIO -> [ESP32] Wi-Fi Marauder -> Sniff [raw]? For me the file gets written in /apps_data/marauder/pcaps with 0 bytes.

I'm running on latest ESP32Marauder with the MultiboardS3 firmware, flashed through FZEasyMarauderFlash that uses esptool since I can't make the ESP32-S3 enter reflash mode, I've tried tons of methods, even the common one, hold the boot button, click the reset button, and release the boot button, nothing works. The `ESP Flasher` Flipper Zero app depends on this functionality to flash a firmware directly through Flipper.

TL;DR, I was able to successfully flash new firmware into your guys's board through esptool, but not through either the esp.huhn.me website or the `ESP Flasher` Flipper app, however, the Multiboard S3 firmware doesn't appear to have serial communication between flipper's SD card and the ESP32-S3 board, thus, not successfully writing .pcap files from sniffing network packets.

![image](https://github.com/ElectronicCats/flipper-shields/assets/17802984/1837bea6-887b-422a-8df2-497256ec3f99)
![image](https://github.com/ElectronicCats/flipper-shields/assets/17802984/74e1fd1e-c5a9-4dea-b0f0-82c9aee7bce9)

**Comment by xpeqex** [comment]
@b1scoito 
Thank you for reporting the issue with the .pcap files.

According to the reports created in the[ justcallmekoko/ESP32Marauder](https://github.com/justcallmekoko/ESP32Marauder) repository, there is a firmware issue that prevents the ESP32 module from communicating with the flipper via UART. This results in the issue that you have reported.

Other Marauder users, regardless of the board, have experienced this error and have shared some of their solutions at the following link:

https://github.com/justcallmekoko/ESP32Marauder/issues/249

Unfortunately, the solutions presented in this link are only functional for ESP32-S2 modules. In these cases, the port of the ESP32-S2 module has been changed to resolve the error.

Our Flipper Addon Marauder includes an ESP32-S3, so it is not compatible with this solution. I have tried this on my own and it does not work.

Here is an issue related to this issue an ESP32-S3 modules: https://github.com/justcallmekoko/ESP32Marauder/issues/370

Because this issue is directly related to the firmware, which is external, we cannot do much more than wait for the firmware developers to find a solution and add ESP32-S3 module compatibility to change its port and thus resolve the issue that .pcap files are not saving any information.

Regarding how to put the module into boot/flash mode, I recommend referring to this section of our wiki:

- [Flipper AddOn: Marauder Spoof update the ESP32-S3 marauder firmware](https://github.com/ElectronicCats/flipper-shields/wiki/Flipper-Add%E2%80%90On:-Marauder-Spoof#update-the-esp32-s3-marauder-firmware)


I hope this information is helpful and we appreciate your understanding.

Greetings

**Comment by b1scoito** [comment]
I've tried to edit the firmware to match the pins present in the board, and enabling the ENABLE_PACKETS_SERIAL macro, but wasn't able to successfully compile it, can I get in touch with you guys so we can take a look at it and create a separate fork for your guys's board?

**Comment by xpeqex** [comment]
@b1scoito 

It is nice to read you tried solving this. 

Regarding the request to do a Marauder fork, we have plans to do it, modify the firmware, and make it compatible with the ESP32-S3 module since we noticed that the firmware is mainly focused on ESP32-S2 modules. There is no estimated time when we will do this, so we kindly recommend you keep posted here in GitHub and the wiki updates.

Additionally, I will keep track of the issue on the official Marauder repo in case there is any update.

**Comment by xpeqex** [comment]
Hello @b1scoito 

I am glad to let you know that our team worked on this and compiled a special firmware `.bin` file in order to make the UART communication work and now the pcaps files are not empty. You can find the file [here]( https://github.com/ElectronicCats/ESP32Marauder/blob/b6e1416794d4332923a48b3fb893e79b4a1a9e15/esp32_marauder/esp32_marauder.ino.esp32s3.bin). 

Please perform the update for the ESP32-S3 module as usual following the [**Update the ESP32-S3** guide](https://github.com/ElectronicCats/flipper-shields/wiki/Flipper-Add%E2%80%90On:-Marauder-Spoof#update-the-esp32-s3-marauder-firmware) swapping the Firmware file with the mentioned on this message.

Let me know if that solves the issue!

**Comment by xpeqex** [comment]
@b1scoito 
No worries, we are also happy that is now working!

Feel free to reach out to us again. 😼

**Comment by b1scoito** [comment]
Hello @sabas1080, I've been trying to compile the latest ESP32Marauder which adds support for selecting the HTML for Evil Portal through serial, I've managed to compile it successfully with all the changes in the `configs.h` file, but pcaps are being written still, with 0 bytes, of course, with your compiled firmware it works, but with mine which is just one version ahead of yours, doesn't.

![image](https://github.com/ElectronicCats/flipper-shields/assets/17802984/2dd2cab9-149e-4a4d-9c5e-3c7ef04d97b2)

**Comment by b1scoito** [comment]
I'm interested in compiling it myself, which changes are needed to be made in the Arduino Core?

**Comment by sabas1080** [comment]
I have uploaded the necessary changes, but still need to do some testing.

**Comment by b1scoito** [comment]
I've tried setting the serial as shown on your commit: https://github.com/ElectronicCats/ESP32Marauder/commit/d40a452e608f7ec0953d48c42cc4eac72f838341 but still, to no success.

**Comment by sabas1080** [comment]
can you write pcap files with information with the current changes from the last commit? I need to know what works for you with the latest changes.

**Comment by xpeqex** [comment]
@b1scoito 
I could not resist joining the conversation. Could you please let me know the saving path for the HTML file, so the Add-On can read it? 🐬

**Comment by b1scoito** [comment]
> can you write pcap files with information with the current changes from the last commit? I need to know what works for you with the latest changes.

I haven't tried compiling your latest commit, but the first version you sent me worked with writing .pcap files.

**Comment by b1scoito** [comment]
> @b1scoito I could not resist joining the conversation. Could you please let me know the saving path for the HTML file, so the Add-On can read it? 🐬

The Wi-FI Marauder updated so we can set the HTML file for Evil Portal through and it works flawlessly. After those changes some stuff changed in the code, but nothing I could see that affects the writing of .pcap files.

@sabas1080 Everything I did was getting the source-code from the latest v1.35.5 ESP32Marauder release and changing the macros in configs.h, and the serial pins, still, setting the HTML file for Evil Portal through serial works but not writing the buffer stream into .pcap files. What have you done in your first release to make it work?

**Comment by sabas1080** [comment]
@b1scoito  I have only changed RX1 and TX1 to pins 17 and 18 according to our schematic, note that you must use the core version 2.0.11
https://github.com/espressif/arduino-esp32/blob/2.0.11/cores/esp32/HardwareSerial.cpp#L58

**Comment by sabas1080** [comment]
@b1scoito Can you test the latest changes in the last commit? https://github.com/ElectronicCats/ESP32Marauder/commit/dcd55e64eb59fe133e635692eb9231adcc4cc120

**Comment by b1scoito** [comment]
Ok. It worked, I see that in the commit before that one the RX and TX pins order is INVERTED! That made me so confused. Now that I'm using the esp32 core 2.0.11 (The ESP32Marauder tutorial says to install 2.0.10 so I had that one installed) and changed the order of the RX and TX pins it worked.

Should we PR to the original ESP32Marauder repository with a EC_SHIELD macro?

**Comment by sabas1080** [comment]
looks good, it would be interesting

**Comment by xpeqex** [comment]
@b1scoito 

We would like to thank you for your comments and contributions to the Electronic Cats - Marauder community.

We are very happy that you have trusted us and are part of our community. Your report encouraged us to fix this, so the capabilities of Marauder Spoof Shield are now unleashed.

We hope that the Marauder team can take a look at your PR and add the Electronic Cats Marauder Add-On to the official repo. We will keep an eye on this.

Once again, we thank you for bringing this issue and your feedback to our attention. Please do not hesitate to contact us again if you have any questions or suggestions.
<p align=right>
<img src="https://github.com/ElectronicCats/flipper-shields/assets/107638696/f412a4ae-147b-426b-8d6e-aa41182e0721" width=250>
</p>
